File: codility/CountDistinctSlices.js
function solution(M, A) {
    // 3 4 5 4 2
    const N = A.length;
    let left = 0;
    const set = new Set();
    let count = 0;
    for (let right = 0; right < N; right++) {
        // right出现重复时，left右移
        while (set.has(A[right])) {
            set.delete(A[left]);
            left++;
        }
        set.add(A[right]);
        count += right - left + 1;
        if (count > 1000000000) {
            return 1000000000;
        }
    }
    return count;
}

File: codility/python/CountDistinctSlices.py
def solution(M, A):
    # Implement your solution here
    n = len(A)
    count = 0
    left = 0
    right = 0
    arr = [0] * (M + 1)
    while left < n:
        while right < n:
            if arr[A[right]] == 1:
                break
            arr[A[right]] = 1
            right += 1
        count += right - left
        if count > 1000000000:
            return 1000000000
        arr[A[left]] = 0
        left += 1
        
    return count